### PR TITLE
Created Separate page to show searched clients

### PR DESCRIPTION
**Pull request by Shannibrst**
Please merge

### CODE REVIEW

File: imports/startup/client/routes/searchClients.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { AppController } from './controllers';
+import { DefaultAdminAccessRoles } from '/imports/config/permissions';
+
+
+Router.route('adminDashboardsearchClientsView', {
+  path: '/clients/view_all/:searchKey',
```
Comment (pgorecki) [inline imports/startup/client/routes/searchClients.js]: use camelCase for routing, change route name to `searchClientsView`

File: imports/ui/clients/searchClient.html [hunk 2]
```diff
@@ -79,4 +79,9 @@ <h4 class="text-center">Pending Clients</h4>
       DOB: {{formatDate dob}}
     </div>
   {{/if}}
+  {{#if viewAll}}
```
Comment (pgorecki) [inline imports/ui/clients/searchClient.html]: add `View more` button at the bottom of the list (in the footer), not for every item

File: imports/ui/clients/searchClient.html [hunk 3]
```diff
@@ -79,4 +79,9 @@ <h4 class="text-center">Pending Clients</h4>
...
+   <div class="text-muted">
+    <a href="javascript:void(0);" id="/clients/view_all/{{searchStr}}" class="btn btn-sm btn-block btn-primary text-center view_all">View All</a>
```
Comment (pgorecki) [inline imports/ui/clients/searchClient.html]: do not use `id` to store urls!

File: imports/ui/clients/searchClient.js [hunk 4]
```diff
@@ -60,10 +60,12 @@ const debouncedSearch = _.debounce((query, options, callback) => {
   Meteor.call('searchClient', query, options, (err, res) => {
     if (err) {
       Alert.error(err);
-    } else {
+    } else { var i=2;
```
Comment (pgorecki) [inline imports/ui/clients/searchClient.js]: there must be a more elegant way to implement this idea. How about returning an object with 2 fields: search results, a flag indicating if a button should be displayed?

File: imports/ui/clients/searchClient.js [hunk 5]
```diff
@@ -146,6 +148,9 @@ Template.searchClient.events(
     'click .client-search-icon-container': () => {
       $('#search-client-keyword').focus();
     },
+	'click .view_all': () => {
+		Router.go($(".view_all").attr('id'));
```
Comment (pgorecki) [inline imports/ui/clients/searchClient.js]: do not use `id` like this

File: imports/ui/clients/viewAllClient.html [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<template name="viewAllClient">
```
Comment (pgorecki) [inline imports/ui/clients/viewAllClient.html]: rename to viewAllClients

File: imports/ui/clients/viewAllClient.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import { Mongo } from 'meteor/mongo';
+//const SearchClient = Meteor.isClient ? new Mongo.Collection('search') : undefined;
+const SearchClient = new Mongo.Collection('search');
```
Comment (pgorecki) [inline imports/ui/clients/viewAllClient.js]: ??? why are you creating mongo collection here? pass search results via router data

File: imports/ui/clients/searchClient.js (hunk 4 above)
Comment (art4ride) [reply imports/ui/clients/searchClient.js]: Also let's keep the `let` and `const` usage favorable to `var`

File: imports/ui/clients/viewAllClient.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import { fullName,viewAll } from '/imports/api/utils';
+import './viewAllClient.html';
+
+const tableOptions = {
+  columns: [
+    {
+      title: 'Client Name',
+	  data: function(row){
+		  const name = (`${row.firstName.trim()} ${row.lastName.trim()}`).trim();
+		  return `<a href="/clients/${row.clientId}?schema=${row.schema}">${name}</a>`;
```
Comment (art4ride) [inline imports/ui/clients/viewAllClient.js]: Let's use `Router` module to generate urls instead of hardcoding them

**Comment by Shannibrst** [comment]
Fixed all issues generated from our end. There are still 4-5 errors which are in other section of code and was already present in master branch.  Do I need to fix these issues as well?

File: imports/ui/clients/viewAllClient.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      data: 'dob',
+      render(value) {
+        return moment(value).format('MM/DD/YYYY');
+      },
+    },
+  ],
+  dom: TableDom,
+};
+
+let tableData = [];
```
Comment (art4ride) [inline imports/ui/clients/viewAllClient.js]: Please store data not in global variable. use reactive var instead.

File: imports/ui/clients/viewAllClient.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+import { ReactiveVar } from 'meteor/reactive-var';
+import './viewAllClient.html';
+
+const tableOptions = {
+  columns: [
+    {
+      title: 'Client Name',
+      data(row) {
+        const name = (`${row.firstName.trim()} ${row.lastName.trim()}`).trim();
+        return `<a href="/clients/${row.clientId}?schema=${row.schema}">${name}</a>`;
```
Comment (art4ride) [inline imports/ui/clients/viewAllClient.js]: Hardcoded url issue still persists. Use `Router.path` function instead

File: imports/ui/clients/searchClient.js [hunk 11]
```diff
@@ -146,6 +150,9 @@ Template.searchClient.events(
     'click .client-search-icon-container': () => {
       $('#search-client-keyword').focus();
     },
+    'click .view_all': () => {
+      Router.go($('.view_all').attr('search-client'));
```
Comment (art4ride) [inline imports/ui/clients/searchClient.js]: instead of doing a global jQuery query you can use event parameter passed to this function already

**Comment by art4ride** [comment]
@Shannibrst don't worry about not PR related errors

**Comment by Shannibrst** [comment]
I have made all changes as mentioned above except 1. We cannot return function of array within template helper function by using reactive var that is why I use tableData variable. I tried all the way to use reactive var within  child function inside tableData() but always got undefined variable.

**Pull request merged**